java/sage/media/bluray/BluRayFile.java

```diff
@@ -0,0 +1,496 @@
+ */
+package sage.media.bluray;
+
+import sage.media.format.ContainerFormat;
+import sage.nio.BufferedFileChannel;
+import sage.nio.LocalFileChannel;
+import sage.nio.RemoteFileChannel;
+import sage.nio.SageFileChannel;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.MappedByteBuffer;
+import java.nio.channels.*;
+
+public class BluRayFile extends FileChannel implements BluRayStreamer, SageFileChannel
+{
+  private final SageFileChannel sageFileChannel;
+  private final long totalSize;
+  private final java.io.File bdmvDir;
+  private final sage.media.bluray.BluRayParser bdp;
+  private final sage.media.bluray.MPLSObject currPlaylist;
+  private final java.io.File[] fileSequence;
+  private final long[] fileOffsets; // bytes
+  private final long[] ptsOffsets; // 45kHz
+  private final long[] chapterOffsets; // 45kHz
+  private int currFileIndex;
+  private final java.io.File streamDir;
+  private int prefAngle;
+  private final int targetTitle;
+
+  /**
+   * Create a new <code>BluRayStreamer</code> for a local file.
+   *
+   * @param bdmvDir The directory containing the bluray.
+   * @param directBuffer Use a direct byte buffer?
+   * @param targetTitle The title to select first.
+   * @param readBufferSize The size to be used for the read buffer.
+   * @throws IOException If there is an I/O related error.
+   */
+  public BluRayFile(java.io.File bdmvDir, boolean directBuffer, int targetTitle, int readBufferSize) throws IOException
+  {
+    this(null, bdmvDir, directBuffer, targetTitle, readBufferSize);
+  }
+
+  /**
+   * Create a new <code>BluRayStreamer</code> for a remote file.
+   *
+   * @param hostname The hostname of the server hosting the bluray content.
+   * @param bdmvDir The directory containing the bluray.
+   * @param directBuffer Use a direct byte buffer?
+   * @param targetTitle The title to select first.
+   * @param readBufferSize The size to be used for the read buffer.
+   * @throws IOException If there is an I/O related error.
+   */
+  public BluRayFile(String hostname, java.io.File bdmvDir, boolean directBuffer, int targetTitle, int readBufferSize) throws IOException
+  {
+    if (hostname == null)
+      // Creates
+      sageFileChannel = new BufferedFileChannel(new LocalFileChannel((String)null, true), readBufferSize, directBuffer);
+    else
+      // Opens a socket, but not any files. We will do that with openFile().
+      sageFileChannel = new BufferedFileChannel(new RemoteFileChannel(hostname, (String)null, true), readBufferSize, directBuffer);
+
+    this.bdmvDir = bdmvDir;
+    int targetTitleTmp = targetTitle;
+    bdp = new sage.media.bluray.BluRayParser(bdmvDir, hostname);
+    bdp.fullyAnalyze();
+    if (targetTitleTmp <= 0)
+      targetTitleTmp = bdp.getMainPlaylistIndex() + 1;
+    this.targetTitle = Math.max(1, Math.min(targetTitleTmp, bdp.getNumPlaylists()));
+    currPlaylist = bdp.getPlaylist(targetTitleTmp - 1);
```

Comment:
To match the old code, this should change (targetTitleTmp - 1) to be (this.targetTitle - 1). May be the same...but rather than digging through it all to analyze, just change this var name. :)